### PR TITLE
Refactor client code

### PR DESCRIPTION
**Pull request by jgillis01**
Refactoring client code and adding additional process check.

### CODE REVIEW

File: lib/poller/client.ex [hunk 1]
```diff
@@ -15,30 +15,42 @@ defmodule Poller.Client do
...
     %__MODULE__{client | current_list: rest}
   end
 
-  defp notify(pid) when is_pid(pid) do
-    case Process.alive?(pid) do
-      true ->
-        Process.send(pid, :call, [])
+  defp get_processes(registry_key) do
+    ProcessRegistry
+    |>Registry.lookup(registry_key)
```
Comment (dbernheisel) [inline lib/poller/client.ex]: spacing

File: lib/poller/client.ex [hunk 2]
```diff
@@ -15,30 +15,42 @@ defmodule Poller.Client do
...
 
     notify_next(%__MODULE__{client | current_list: processes})
   end
 
   def notify_next(client) do
     [process | rest] = client.current_list
-    notify(process)
+
+    process
+    |> valid_process?(client.registry_key)
```
Comment (dbernheisel) [inline lib/poller/client.ex:28]: 👍

File: lib/poller/client.ex [hunk 3]
```diff
@@ -15,30 +15,42 @@ defmodule Poller.Client do
...
+
+    registered =
+      registry_key
+      |> get_processes()
+      |> Enum.any?(& (&1 == process))
 
-      _ ->
-        :ok
-    end
+    alive && registered
```
Comment (dbernheisel) [inline lib/poller/client.ex]: What do you think of short-circuiting with a `with` statement so it'll return false if the process isn't alive? That way you can skip checking if it's a registered process.

```elixir
with true <- Process.alive?(process),
     processes <- get_processes(registry_key) do
  Enum.any?(processes, & &1 == process)
end
```

**Review by dbernheisel — APPROVED** [review]

File: lib/poller/client.ex (hunk 1 above)
Comment (jgillis01) [reply lib/poller/client.ex]: Thank you!

File: lib/poller/client.ex (hunk 3 above)
Comment (jgillis01) [reply lib/poller/client.ex]: Sounds good to me.  Thank you

**Pull request merged**